app/src/main/java/com/sdp13epfl2021/projmag/database/Utils.kt

```diff
@@ -1,9 +1,43 @@
 package com.sdp13epfl2021.projmag.database
 
-import com.google.firebase.firestore.FirebaseFirestore
+import android.content.Context
+import com.google.firebase.auth.ktx.auth
+import com.google.firebase.firestore.ktx.firestore
+import com.google.firebase.ktx.Firebase
+import com.google.firebase.storage.ktx.storage
+import java.io.File
 
 object Utils {
 
-    val projectsDatabase = CachedProjectsDatabase(FirebaseProjectsDatabase(FirebaseFirestore.getInstance()))
+    lateinit var projectsDatabase: CachedProjectsDatabase
+    lateinit var userDatabase: UserDataDatabase
+    lateinit var fileDatabase: FileDatabase
 
+    private var initalized = false
+
+    @Synchronized
+    fun init(context: Context) {
+        if (!initalized) {
+            val firestore = Firebase.firestore
+
+            // Uncomment to disable firebase cache
+            /* val settings = firestoreSettings {
+                isPersistenceEnabled = false
+            }
+            firestore.firestoreSettings = settings*/
+
+            projectsDatabase =
+                CachedProjectsDatabase(
+                    OfflineProjectDatabase(
+                        FirebaseProjectsDatabase(
+                            firestore
+                        ),
+                        File(context.filesDir, "projects")
+                    )
+                )
+            fileDatabase = FirebaseFileDatabase(Firebase.storage, Firebase.auth)
+            userDatabase = UserDataFirebase(firestore, Firebase.auth)
+            initalized = true
+        }
+    }
 }
```

Comment:
Maybe it would be safer, and would also make more sense, to have a `class` instead of an object if we have an `init` method.

Imagine we keep the `context` between activities, this would not be expected.
However, if we instantiate an object an object with `context`, it would stay within the `activity` and not share the `context` with another one.
However, you can still put static stuff in a `companion object`

Reply:
I changed the Utils object to a class and its constructor ask for a context.